src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/Receivers.kt

```diff
@@ -181,36 +262,38 @@ class MarkdownTaskReceiver(
 
     private fun getCmdLineArguments(request: CompileRequest): Map<String, String> {
         val outputFileName = Files.getNameWithoutExtension(request.mainFileName) + ".tex"
-        val projTasks = this.tasksDir.resolve(request.id)
-        val mainFile = request.mainFileName
-        val outputFile = projTasks.resolve(outputFileName)
-        val projectRootAbsPath = this.tasksDir.toAbsolutePath().parent
-
-        val values = listOf(projectRootAbsPath.toString(), "",
-                mainFile.toString(), outputFile.toString(), PANDOC_DEFAULT_FONT)
+        val projTasks = this.tasksDir.resolve(request.id).resolve("files")
+        val mainFile = StringUtils.stripStart(request.mainFileName, "/")
+        val values = listOf(projTasks.toString(), "",
+                mainFile, outputFileName, PANDOC_DEFAULT_FONT)
 
-        return (arguments zip values).map { it.first to it.second }.toMap()
+        return (arguments zip values).map { it.first to "\"${it.second}\"" }.toMap()
     }
 
     // converts Markdown into tex via pandoc
     private fun convertMarkdown(commandArguments: Map<String, String>): Int {
         val substitutor = StringSubstitutor(commandArguments)
-        val rawCommandLine = config.getString(COMPILE_COMMAND_KEY)
+
+        // the '\' exists because of the double slash in the config,
+        // where in turn $ should be protected, so we remove the value here
+        val rawCommandLine = config.getString(COMPILE_COMMAND_KEY).replace("\\", "")
         val commandLine = substitutor.replace(rawCommandLine)
         return runCommandLine(commandLine)
     }
 
     private fun compileTex(request: CompileRequest, tex: File): CompileResponse {
-        request.toBuilder().setMainFileName(tex.name).build()
-
         val targetTex = FileDto
                 .newBuilder()
                 .setName(tex.name)
-                .setContents(
-                        ByteString.copyFrom(FileUtils.readFileToByteArray(tex)))
                 .build()
 
-        request.fileRequest.toBuilder().addFile(targetTex)
-        return texCompiler.compile(request)
+        val texRequest = 
+                request.toBuilder()
+                       .setMainFileName(tex.name)
```

Comment:
File.name and FileDto.name are different sorts of things. The former is a short name inside file's directory. The latter is an absolutive name: relative to project root, prefixed with /.